go-packages/meep-subscriptions/websocket.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2021  InterDigital Communications, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package subscriptions
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	log "github.com/InterDigitalInc/AdvantEDGE/go-packages/meep-logger"
+	"github.com/gorilla/websocket"
+)
+
+type WebsocketCfg struct {
+	Timeout time.Duration `json:"timeout"`
+}
+
+type Websocket struct {
+	Cfg               *WebsocketCfg
+	Id                string                                       `json:"id"`
+	State             string                                       `json:"state"`
+	Endpoint          string                                       `json:"endpoint"`
+	Uri               string                                       `json:"uri"`
+	ConnectionHandler func(w http.ResponseWriter, r *http.Request) `json:"-"`
+	Connection        *websocket.Conn                              `json:"-"`
+	MsgHandler        chan []byte                                  `json:"-"`
+	Done              chan int                                     `json:"-"`
+}
+
+const (
+	WsStateInit  = "Init"
+	WsStateReady = "Ready"
+)
+
+// Websocket connection upgrader
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+	CheckOrigin:     func(r *http.Request) bool { return true },
+}
+
+func newWebsocket(cfg *WebsocketCfg) (*Websocket, error) {
+	// Create new websocket
+	var ws Websocket
+	ws.Cfg = cfg
+
+	// Generate a random websocket URI
+	randomStr, err := generateRand(12)
+	if err != nil {
+		log.Error(err.Error())
+		return nil, err
+	}
+	ws.Id = randomStr
```

Comment:
For the sake of clarity, I would pre-pend the websocket endpoint with `websocket-` (ex: `websocket-12345abcd`)
This way, it would be clearer what this id is (ex. when shown in the MEC Sandbox console)